winpr/libwinpr/utils/asn1/asn1.c

```diff
@@ -1020,17 +1020,24 @@ static size_t WinPrAsn1DecReadIntegerLike(WinPrAsn1Decoder* dec, WinPrAsn1_tag e
 	size_t ret = readTagAndLen(dec, &dec->source, &tag, &len);
 	if (!ret || (tag != expectedTag))
 		return 0;
-	if (!Stream_CheckAndLogRequiredLength(TAG, &dec->source, len) || (len > 4))
+	if (len == 0 || !Stream_CheckAndLogRequiredLength(TAG, &dec->source, len) || (len > 4))
```

Comment:
@fifthdegree sure that a 0 length is invalid? should that not be 0 like before this change?

Reply:
Yes.

>8.3 Encoding of an integer value
8.3.1 The encoding of an integer value shall be primitive. The contents octets shall consist of one or more octets.

Reply:
@fifthdegree thx, was unsure if 0 is allowed or not, that clarifies it.